### DISCUSSION TITLE
GitHub website became grey with black font.

### DISCUSSION THREAD

**Discussion by vladislavpyatnitskiy** [community/community#113514] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

I am experiencing issue with GitHub. Website became grey and font became black. The problem exists only on Safari on Mac, other browsers and devices do not have issue. What can be done or where should i submit my problem? I have been looking for issue solution but now stuck. I have attached the screen of GitHub below
<img width="1098" alt="Screenshot for GitHub" src="https://github.com/community/community/assets/111236801/1e6466b8-7395-4a3f-ac3e-642ba2d22fe2">

Thank you in advance!

**Comment by Dazcam** [comment]
Also experiencing this. Started yesterday 20/03/24

**Comment by Torrekie** [comment]
They abandoned our Safari users :(

**Comment by git-sumana** [comment]
Clear the cache and then try it again buddy, its working for me

**Reply by vladislavpyatnitskiy** [reply]
I tried via option + command + E, but it did not work.

**Reply by git-sumana** [reply]
Did you try signing out from github and coming back?

**Reply by git-sumana** [reply]
<img width="959" alt="ss" src="https://github.com/community/community/assets/137371757/92a9d58d-584c-4e54-bf6a-bd71ffebc390">

**Reply by vladislavpyatnitskiy** [reply]
I have also tried to sign out but nothing changed.

**Comment by Tgentil** [comment]
The issue with GitHub appearing grey with black font on Safari could be due to browser settings or GitHub theme preferences. 

First, check your GitHub theme settings in your [GitHub settings](https://github.com/settings/appearance). GitHub offers themes like light, dark, and dim modes, and switching them can change the website's appearance. 

If the theme is not the issue, try these Safari troubleshooting steps:
1. **Check Safari's Preferences and Extensions**: Ensure no extensions are altering website appearances.
2. **Reset Safari Settings**: Sometimes, resetting to default can help.
3. **Update Safari**: Ensure you're using the latest version.
4. **Private Browsing in Safari**: This helps determine if the issue is related to browser data or settings.

If these steps don't help, consider contacting [GitHub Support](https://support.github.com/contact).

**Comment by mccarthy-m-g** [comment]
I started experiencing this recently as well on Safari Version 14.1.2 (14611.3.10.1.7), and other parts of the site are also broken (e.g., most parts of the navbar, comment boxes, etc.). Assumed this was due to them dropping support for older versions of Safari with all the UX changes they've been making...

@vladislavpyatnitskiy and @Dazcam, are you also using older versions of Safari?

**Comment by Torrekie** [comment, marked as answer]
All Safari versions shipped by pre Darwin 20 (iOS 14, macOS 11) has been affected by this, GitHub was suddenly updating something that not supported by our Safari.

**Comment by jesusburnsneon** [comment]
Noticed it today right after signing on to Github with Safari.  Hopefully this gets fixed soon.

**Comment by nieto915** [comment]
its happening to me now its crazy. 
guess im a chrome guy now.

**Comment by nomakewan** [comment]
Can confirm that on iOS 14.3, neither Safari nor Chrome can display Github properly. The background is black and the text is black, making it impossible to use.

![IMG_7329s](https://github.com/community/community/assets/11485516/f33ccaa4-54e1-48f9-b824-87b4adcf555c)
![IMG_7331s](https://github.com/community/community/assets/11485516/5e2199cd-4632-432e-9f4e-39b885c68199)

**Comment by Torrekie** [comment]
Now it affects all pre Darwin 21 (iOS 15 / macOS 12) WebKit, what da hell GitHub doing?

**Comment by jasperJNS** [comment]
workaround is to go to https://github.com/settings/appearance and use the light tritanopia setting (in the night theme). its still clearly bugged but stuff is visible at least.

**Comment by hitmanex** [comment]
@nomakewan disable dark mode on iOS. It’s some bug with dark mode

**Reply by nomakewan** [reply]
And yet this was never an issue before. In addition, I'm not going to enable eye-blinding mode on my phone just to be able to use Github on mobile.

Thank you for the suggestion, however. It may help others who are willing to switch.

**Reply by hitmanex** [reply]
If you’re logged in go to https://github.com/settings/appearance 
Change it from “sync with system” to “light default”..or maybe the others will work too

**Reply by nomakewan** [reply]
I was not logged in (since the login is also blank) but I will try that. As long as doing that doesn't mess up my appearance settings on desktop, might be a valid workaround. Thanks!

**Comment by vladislavpyatnitskiy** [comment]
I have updated my Mac and everything started to work again as has to.

**Comment by ATHARV-NALE** [comment]
i ma also expering same on chrome browser

**Comment by Torrekie** [comment]
Please reopen this, updating system should not be the only solution

**Reply by vladislavpyatnitskiy** [reply]
Done

**Comment by nomakewan** [comment]
Since this is reopened, I would like to update this to say that changing to light mode is no longer a valid fix either on iOS 14.3. Not only is the text black-on-black when in dark mode, but within the last few months yet another change was made that now makes all buttons nonfunctional bubble buttons. So Github is completely useless on these browsers even if you _could_ read the text. And if you hadn't acquired the Github app prior to it too being updated to require iOS 16, then tough nuts on being able to interact with the platform in any way.

Here's what Github looks like now (when not logged in) on iOS 14.3:

![IMG_E8357](https://github.com/user-attachments/assets/76f3432f-8dc5-46f9-996a-18b9426ef04b)

Entirely unacceptable. For what reason were these changes even made?

**Comment by Torrekie** [comment]
GitHub now uses some quite new ECMAScript features that not supported by Safari 14.x, which now affects all versions before iOS 15.4 and macOS 11.x (Maybe also 12.x).

For iOS, PoomSmart wrote a workaround tweak https://github.com/PoomSmart/GitHubWebLegacyCompat, which at least fixed this terrible black screen, this tweak can be rewritten to Safari extension
<img width="1440" alt="image" src="https://github.com/user-attachments/assets/23e5b8e5-6350-4d7d-b0b7-1fd24e99b72e" />

**Reply by nomakewan** [reply]
Just tested the tweak and it does fix the text, but sadly as expected it does not fix the useless buttons. So you still can't do basic things like change branches or grab code.

Thanks!

EDIT: Opened an issue on the repo, it seems active so maybe we'll have a solution soon!

**Reply by Torrekie** [reply]
Another way is to compile a newer webkit like https://github.com/UInt2048/CyberKit, but CyberKit performs really bad and out of maintenance, so theres really need someone to take this project again
